api/pkg/transformer/feast/call.go

```diff
@@ -65,13 +65,23 @@ func (fc *call) do(ctx context.Context, entityList []feast.Row, features []strin
 	return callResult{tableName: tableName, featureTable: featureTable, err: nil}
 }
 
+func getFeatureTypeMapping(featureTableSpec *spec.FeatureTable) map[string]types.ValueType_Enum {
+	mapping := make(map[string]types.ValueType_Enum, len(featureTableSpec.Features))
+	for _, feature := range featureTableSpec.Features {
+		feastValType := types.ValueType_Enum(types.ValueType_Enum_value[feature.ValueType])
+		mapping[feature.Name] = feastValType
+	}
+	return mapping
+}
+
 // processResponse process response from feast serving and create an internal feature table representation of it
 func (fc *call) processResponse(feastResponse *feast.OnlineFeaturesResponse) (*internalFeatureTable, error) {
 	responseStatus := feastResponse.Statuses()
 	responseRows := feastResponse.Rows()
 	entities := make([]feast.Row, len(responseRows))
 	valueRows := make([]transTypes.ValueRow, len(responseRows))
 	columnTypes := make([]types.ValueType_Enum, len(fc.columns))
+	columnTypeMapping := getFeatureTypeMapping(fc.featureTableSpec)
```

Comment:
Shall we cache it as a struct field so that it wont be recomputed for each call?

Reply:
updated